lib/greenbar/tags/join.ex

```diff
@@ -52,19 +52,16 @@ defmodule Greenbar.Tags.Join do
                               {false, scope}
                           end
     joiner = get_attr(attrs, "with", ", ")
-    case get_remaining(scope, remaining_key, attrs) do
+    case get_remaining(scope, id, remaining_key, attrs) do
       nil ->
         {:halt, scope}
       [] ->
-        scope = scope
-        |> Scoped.erase(downstream_key)
-        |> Scoped.erase(remaining_key)
-        {:halt, scope}
+        {:halt, put(scope, id, remaining_key, [])}
       [h|t] ->
         var_name = get_attr(attrs, "as", "item")
         child_scope = new_scope(scope)
-        {:ok, child_scope} = Scoped.set(child_scope, var_name, h)
-        {:ok, scope} = set_remaining(scope, remaining_key, t)
+        child_scope = put(child_scope, :global, var_name, h)
+        scope = put(scope, id, remaining_key, t)
 
```

Comment:
Seems like these scope operations should be handled by `Scoped`. `Scoped.set_global` instead of `put(scope, :global, ...)`, etc.